server.go

```diff
@@ -163,21 +163,52 @@ func (s *server) UpdatePin(ctx context.Context, request v0.UpdatePinRequestObjec
 		user = usersResponse.Payload[0]
 	}
 
+	var results []v0.UpdatePinResponseObject
+	g, ctx := errgroup.WithContext(ctx)
+
+	for i := range s.cfg.Nuki.SmartLockDevices {
+		func(smd int64) {
+			g.Go(func() error {
+				res, err := s.updateSmartLock(ctx, request, smd, user)
+				results = append(results, res)
```

Comment:
i think this is not thread safe. Better use a channel. Or check the http status code within the `Go` func and return an error if not OK. Like the check from line 187 could also happen here. 
```suggestion
  if err != nil {
    return err 
  }
  // the casting and stuff 
```

Reply:
Not sure if the updatePin function is so nice to use, because the caller has to cast its results

Reply:
I think updatePin should maybe only return an error and not a (v0)Response type/

The caller has to put the result into the correct v0(type).

This way it is easier to call updatePin without having to cast it.